src/dotnet/APIView/APIViewWeb/Repositories/ReviewManager.cs

```diff
@@ -380,10 +381,23 @@ public async Task<ReviewModel> CreateMasterReviewAsync(ClaimsPrincipal user, str
                 }
 
                 var renderedCodeFile = new RenderedCodeFile(codeFile);
-                var noDiffFound = await IsReviewSame(review.Revisions.LastOrDefault(), renderedCodeFile);
-                if (noDiffFound)
+                // We should compare against only latest revision when calling this API from scheduled CI runs
+                // But any manual pipeline run at release time should compare against all approved revisions to ensure hotfix release doesn't have API change
+                // If review surface doesn't match with any approved revisions then we will create new revision if it doesn't match pending latest revision
+                if (compareAllRevisions)
                 {
-                    // No change is detected from last revision so no need to update this revision
+                    foreach (var approvedRevision in review.Revisions.Where(r => r.IsApproved).Reverse())
+                    {
+                        if (await IsReviewSame(approvedRevision, renderedCodeFile))
+                        {
+                            return approvedRevision;
+                        }
+                    }
+                }
+
+                if (await IsReviewSame(lastRevision, renderedCodeFile))
```

Comment:
else?

Reply:
createNewRevision is set to true by default to ensure new revision is created if reviews are not same.